plugins/inputs/dropwizard/dropwizard.go

```diff
@@ -0,0 +1,350 @@
+	Version    string               `json:"version"`
+	Gauges     map[string]gauge     `json:"gauges"`
+	Counters   map[string]counter   `json:"counters"`
+	Histograms map[string]histogram `json:"histograms"`
+	Meters     map[string]meter     `json:"meters"`
+	Timers     map[string]timer     `json:"timers"`
+}
+
+// Gathers data from a particular URL
+// Parameters:
+//     acc    : The telegraf Accumulator to use
+//     url    : endpoint to send request to
+//
+// Returns:
+//     error: Any error that may have occurred
+func (d *Dropwizard) gatherURL(
+	acc telegraf.Accumulator,
+	url string,
+) error {
+	now := time.Now()
+
+	resp, err := d.client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	metrics, err := d.DecodeJSONMetrics(bodyBytes)
+	if err != nil {
+		return err
+	}
+
+	// tags can be specified in config either globally or per input
+	// through built-in functionality
+	var tags map[string]string = nil
+
+	//TODO do we need to call SetPricision and is this related to units?
+	
+	for name, g := range metrics.Gauges {
+		if g.Value.Type == IntType {
+			acc.AddGauge(name,
+				map[string]interface{}{ "value": g.Value.IntValue },
+				tags,
+				now)
+		} else if g.Value.Type == FloatType {
+			acc.AddGauge(name,
+				map[string]interface{}{ "value": g.Value.FloatValue },
+				tags,
+				now)
+		}
+	}
+
+	for name, c := range metrics.Counters {
+		acc.AddCounter(name,
+			map[string]interface{}{ "count": c.Count },
+			tags,
+			now)
+	}
+
+	for name, h := range metrics.Histograms {
+		acc.AddHistogram(name,
+			map[string]interface{}{ 
+				"count": h.Count,
+				"max": h.Max,
+				"mean": h.Mean,
+				"min": h.Min,
+				"p50": h.P50,
+				"p75": h.P75,
+				"p95": h.P95,
+				"p98": h.P98,
+				"p99": h.P99,
+				"p999": h.P999,
+				"stddev": h.Stddev,
+			},
+			tags,
+			now)
+	}
+
+	//TODO what to do with the Units?
+	for name, m := range metrics.Meters {
+		acc.AddHistogram(name,
+			map[string]interface{}{ 
+				"count": m.Count,
+				"m15_rate": m.M15Rate,
+				"m1_rate": m.M1Rate,
+				"m5_rate": m.M5Rate,
+				"mean_rate": m.MeanRate,
+			},
+			tags,
+			now)
+	}
+
+	//TODO what to do with duration and rate units?
+	for name, t := range metrics.Timers {
+		acc.AddFields(name,
+			map[string]interface{}{ 
+				"count": t.Count,
+				"max": t.Max,
+				"mean": t.Mean,
+				"min": t.Min,
+				"p50": t.P50,
+				"p75": t.P75,
+				"p95": t.P95,
+				"p98": t.P98,
+				"p99": t.P99,
+				"p999": t.P999,
+				"stddev": t.Stddev,
+				"m15_rate": t.M15Rate,
+				"m1_rate": t.M1Rate,
+				"m5_rate": t.M5Rate,
+				"mean_rate": t.MeanRate,
+			},
+			tags,
+			now)
+	}
+
+	return nil
+}
+
+func init() {
+	inputs.Add("dropwizard", func() telegraf.Input {
+		return &Dropwizard{
+			Timeout: internal.Duration{Duration: time.Second * 5},
+		}
+	})
+}
+
+func (*Dropwizard) DecodeJSONMetrics(jsonInput []byte) (metrics, error) {
```

Comment:
I would suggest instead using json.NewDecoder(reader).Decode() and have DecodeJSONMetrics() take in io.Reader instead of the byte array to save some memory copying